src/combinatorics/simple_graph/density.lean

```diff
@@ -0,0 +1,251 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.basic
+import order.partition.finpartition
+
+/-!
+# Edge density
+
+This file defines the number and density of edges of a relation/graph.
+
+## Main declarations
+
+Between two finsets of vertices,
+* `relation.interedges`: Finset of edges of a relation.
+* `relation.edge_density`: Edge density of a relation.
+* `simple_graph.interedges`: Finset of edges of a graph.
+* `simple_graph.edge_density`: Edge density of a graph.
+-/
+
+open finset
+
+variables {ι κ α β : Type*}
+
+/-! ### Density of a relation -/
+
+namespace relation
```

Comment:
Put this under the namespace `rel` (see https://leanprover-community.github.io/mathlib_docs/data/rel.html#rel)

Reply:
I do not mind either way, but please consider the reason of my choice:
* Even though they are defeq, this is not about `rel` but about `α → β → Prop`.
* The `relation` namespace is already used for things like [`relation.refl_trans_gen`](https://leanprover-community.github.io/mathlib_docs/logic/relation.html#relation.refl_trans_gen) and [`relation.map`](https://leanprover-community.github.io/mathlib_docs/logic/relation.html#relation.map),

Maybe the situation is similar to `function` vs `set`? Let me also plug #1496.

Reply:
Then I weaken my comment - put it under `rel` if you think anyone will ever use projection notation for these declarations. Otherwise, both sound good.

Reply:
Yeah I don't think this will happen, but I will still rename because the namespace is shorter. We're not yet clear as to how to deal with relations.